### PR TITLE
Cleanup downloads folder

### PR DESCRIPTION
**Pull request by asolntsev**
an alternative implementation for https://github.com/selenide/selenide/pull/1249.

Probably I like it more.

### CODE REVIEW

File: src/main/java/com/codeborne/selenide/impl/FileHelper.java [hunk 1]
```diff
@@ -53,9 +54,25 @@ public static File ensureFolderExists(File folder) {
...
+      throw new IllegalStateException("Failed to move file " + srcFile.getAbsolutePath() +
+        " to " + destFile.getAbsolutePath(), e);
+    }
+  }
+
+  public static void cleanupFolder(File folder) {
+    try {
       cleanDirectory(folder);
+      if (folder.isDirectory()) {
+        cleanDirectory(folder);
```
Comment (dstekanov) [inline src/main/java/com/codeborne/selenide/impl/FileHelper.java:71]: why two times call `cleanDirectory(folder)` ?

Comment (asolntsev) [reply src/main/java/com/codeborne/selenide/impl/FileHelper.java:71]: ups. Fixed.

File: src/main/java/com/codeborne/selenide/UniqueDownloadsFolder.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+import com.codeborne.selenide.impl.FileHelper;
+
+import java.io.File;
+
+/**
+ * A unique folder per browser.
+ * It effectively means that Selenide can delete all files in this folder before starting every new download.
+ */
+public class UniqueDownloadsFolder extends DownloadsFolder {
```
[the remark this answers is not in the document]
Comment (asolntsev) [reply src/main/java/com/codeborne/selenide/UniqueDownloadsFolder.java:11]: @dstekanov Class name should not start with verb ("clear").  
Then probably `ManageableDownloadsFolder` is a better name.

[the remark this answers is not in the document]
Comment (asolntsev) [reply src/main/java/com/codeborne/selenide/UniqueDownloadsFolder.java:11]: @dstekanov 
The idea that this folder belongs only to one browser, and no any other browser will ever use this folder.  
This gives us possibility to clean up this folder.  

Other possible names:
* `BrowserDownloadsFolder`
* `TemporaryDownloadsFolder`
* `PerBrowserDownloadsFolder`
* `OneTimeDownloadsFolder`
* `DropOffDownloadsFolder`

[the remark this answers is not in the document]
Comment (SeleniumTestAB) [reply src/main/java/com/codeborne/selenide/UniqueDownloadsFolder.java:11]: I would say the best is BrowserDownloandsFolder. 
TemporaryFolder indicates it is folder for all of the browser, OneTimeFolder indicates it does only exist one time (which is not true), DropOff is not clear, like from what it drops off and downloands?

There is PerBrowserDLFolder but i think it has the same meaning as BrowserDownloandsFolder. 
I think you could also use UniqueBrowserFolder as name but i am not sure if it doesnt hide its intent.

[the remark this answers is not in the document]
Comment (asolntsev) [reply src/main/java/com/codeborne/selenide/UniqueDownloadsFolder.java:11]: Yes, but... `BrowserDownloadsFolder` doesn't say that this folder will be deleted after closing the browser. 
In that sense, `PerBrowserDownloadsFolder` or `BrowserInstanceDownloadsFolder` is probably better?..

**Pull request merged**